pkg/webhook/pod/pods.go

```diff
@@ -237,28 +308,42 @@ func (pc *PodAdmissionControl) AdmitCreatePods(ar *admission.AdmissionRequest) *
 		return util.ARSuccess()
 	}
 
-	tcName, exist := pod.Labels[label.InstanceLabelKey]
-	if !exist {
-		return util.ARSuccess()
-	}
-
-	tc, err := pc.operatorCli.PingcapV1alpha1().TidbClusters(namespace).Get(tcName, metav1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
+	var ownerTc *v1alpha1.TidbCluster
+	var err error
+	if l.IsTidbClusterPod() {
+		tcName, exist := pod.Labels[label.InstanceLabelKey]
+		if !exist {
 			return util.ARSuccess()
 		}
-		klog.Errorf("failed get tc[%s/%s],refuse to create pod[%s/%s],%v", namespace, tcName, namespace, name, err)
-		return util.ARFail(err)
-	}
-
-	if memberUtils.NeedForceUpgrade(tc) {
-		klog.Infof("tc[%s/%s] is force upgraded, admit to create pod[%s/%s]", namespace, tcName, namespace, name)
+		ownerTc, err = pc.operatorCli.PingcapV1alpha1().TidbClusters(namespace).Get(tcName, metav1.GetOptions{})
+		if err != nil {
+			klog.Errorf("failed get tc[%s/%s],refuse to create pod[%s/%s],%v", namespace, tcName, namespace, name, err)
+			return util.ARFail(err)
+		}
+	} else if l.IsGroupPod() {
+		tgName, exist := pod.Labels[label.InstanceLabelKey]
+		if !exist {
+			return util.ARSuccess()
+		}
+		tg, err := pc.operatorCli.PingcapV1alpha1().TiKVGroups(namespace).Get(tgName, metav1.GetOptions{})
+		if err != nil {
+			klog.Errorf("failed get tikvgroup[%s/%s],refuse to create pod[%s/%s],%v", namespace, tgName, namespace, name, err)
+			return util.ARFail(err)
+		}
+		tcName := tg.Spec.ClusterName
+		ownerTc, err = pc.operatorCli.PingcapV1alpha1().TidbClusters(namespace).Get(tcName, metav1.GetOptions{})
+		if err != nil {
+			klog.Errorf("failed get tc[%s/%s],refuse to create pod[%s/%s],%v", namespace, tcName, namespace, name, err)
+			return util.ARFail(err)
+		}
```

Comment:
I think there is no need to check tc for the pod of tikvgroup creation, no need to mix them up, even if tc is good here and the tikvs are created, it is still possible that at some time the PD cluster is down, which is similar with the case that we start the tikvs here even if we get tc failed.

Reply:
We didn't "check" tc here. we are going to fetch cluster level spec information here.